src/mpyl/run_plan.py

```diff
@@ -25,49 +25,77 @@
     ) -> "RunPlan":
         return cls(
-            all_known_projects=all_known_projects, full_plan=plan, selected_plan=plan
+            all_known_projects=all_known_projects, _full_plan=plan, selected_plan=plan
         )
 
-    def select_stage(self, stage: Stage) -> "RunPlan":
+    def select_stage(self, stage_name: str) -> "RunPlan":
+        stage = None
+        for s in self._get_all_stages(use_full_plan=False):
+            if s.name == stage_name:
+                stage = s
+                break
+        if not stage:
+            raise ValueError(
+                f"Unable to select stage outside of the run plan: '{stage_name}'"
+            )
+
         return RunPlan(
             all_known_projects=self.all_known_projects,
-            full_plan=self.full_plan,
+            _full_plan=self._full_plan,
             selected_plan={stage: self.get_projects_for_stage(stage)},
         )
 
-    def select_projects(self, projects: set[Project]) -> "RunPlan":
-        selected_plan = {}
+    def select_project(self, project_name: str) -> "RunPlan":
+        project = None
+        for p in self._get_all_projects(use_full_plan=False):
+            if p.name == project_name:
+                project = p
+                break
+        if not project:
+            raise ValueError(
+                f"Unable to select project outside of the run plan: '{project_name}'"
+            )
 
-        for stage, executions in self.selected_plan.items():
-            selected_plan[stage] = {e for e in executions if e.project in projects}
+        selected_plan = {}
+        for stage in self._get_all_stages(use_full_plan=False):
+            filtered = {
+                e
+                for e in self.get_projects_for_stage(stage, use_full_plan=False)
+                if e.name == project_name
+            }
+            if filtered:
+                selected_plan[stage] = filtered
 
         return RunPlan(
             all_known_projects=self.all_known_projects,
-            full_plan=self.full_plan,
+            _full_plan=self._full_plan,
             selected_plan=selected_plan,
         )
 
-    def has_projects_to_run(
-        self, include_cached_projects: bool, use_full_plan: bool = False
-    ) -> bool:
+    def has_projects_to_run(self, include_cached_projects: bool) -> bool:
         return any(
             include_cached_projects or not project_execution.cached
-            for project_execution in self.get_all_projects(use_full_plan)
+            for project_execution in self._get_all_projects(use_full_plan=False)
         )
 
-    def get_all_projects(self, use_full_plan: bool = False) -> set[ProjectExecution]:
+    def _get_all_stages(self, use_full_plan: bool = False) -> set[Stage]:
+        if use_full_plan:
+            return set(self._full_plan.keys())
+        return set(self.selected_plan.keys())
+
+    def _get_all_projects(self, use_full_plan: bool = False) -> set[ProjectExecution]:
```

Comment:
`_get_all_project_executions`?

Reply:
This method already existed, I just added the private `_` indicator behind it. I do like that name better though 👍